### PR TITLE
Check TRNSYS constants limits

### PR DESCRIPTION
**Pull request by PuaYongHao**
Check if TRNSYS constants limits are exceeded #79

### CODE REVIEW

**Comment by PuaYongHao** [comment]
This needs https://github.com/SPF-OST/pytrnsys_gui/pull/267

File: pytrnsys/trnsys_util/buildTrnsysDeck.py [hunk 1]
```diff
@@ -348,4 +343,42 @@ def replaceLines(self, replaceList):
...
+                maxLineWidth = 0
+
+            for constant in constantsToCheck:
+                match = _re.search(fr"^\b{constant}\s*\d+\b", line, _re.MULTILINE)
+                if match:
+                    if constant == "UNIT":
+                        numOfTrnsysConstants[constant] = numOfTrnsysConstants.get(constant, 0) + 1
+                    elif constant == "CONSTANTS":
+                        split = match.group().split()
+                        numOfTrnsysConstants["EQUATIONS"] = numOfTrnsysConstants.get("EQUATIONS", 0) + int(split[1])
```
Comment (zuckerruebe) [inline pytrnsys/trnsys_util/buildTrnsysDeck.py:367]: "CONSTANTS" vs "EQUATIONS"...isn't there a mismatch?

**Review by zuckerruebe — CHANGES_REQUESTED** [review]
Is this merge request still relevant?

File: pytrnsys/trnsys_util/buildTrnsysDeck.py (hunk 1 above)
Comment (PuaYongHao) [reply pytrnsys/trnsys_util/buildTrnsysDeck.py:367]: > "CONSTANTS" vs "EQUATIONS"...isn't there a mismatch?

If I am not mistaken, "CONSTANTS" is considered as "EQUATIONS" as per discussion with @martin-neugebauer.

Comment (martin-neugebauer) [reply pytrnsys/trnsys_util/buildTrnsysDeck.py:367]: Indeed the TrnsysConstant `nMaxEquations` specifies the maximum for the sum of `EQUATIONS` and `CONSTANTS`, so this approach is correct.

Comment (zuckerruebe) [reply pytrnsys/trnsys_util/buildTrnsysDeck.py:367]: ok, but shouldn't it be something like `elif constant in ("CONSTANTS", "EQUATIONS")` then?

Comment (martin-neugebauer) [reply pytrnsys/trnsys_util/buildTrnsysDeck.py:367]: I'm lacking a bit the code overview in this particular location, but both `EQUATIONS` as well as `CONSTANTS` should be added to the same counter. So if `EQUATIONS` is not added somewhere else, then I agree with @zuckerruebe. One other thing: Why is `UNIT` added to `constant`?

**Pull request merged**